Services/communication_service.c

```diff
@@ -237,9 +253,89 @@ SAT_returnState communication_service_app(csp_packet_t *packet) {
       }
       break;
 
+      case SET_FREQ:
+          cnv8_F(&packet->data[IN_DATA_BYTE], &S_config.freq);
+          //S_config.freq = (float) packet->data[IN_DATA_BYTE];
+          S_config.freq = csp_ntohflt(S_config.freq);
+          status = HAL_S_setFreq(S_config.freq);
```

Comment:
For some reason, this part does not function properly unlike the rest of the set functions. My guess is that here we have to use cnv function but in the rest, we directly assign the value in the packet to the parameters. In `HAL_S_setFreq()` some nonsense values are sent even though `S_config.freq` is fine before going into the function. Using breakpoints, I realized that when you set the frequency, the `S_config.freq` would keep its value until something happens. This is not the case for the others. So, this part needs a fix that I've spent some time figuring out but haven't been successful. It could be because of the way I've defined my structs, too.

Reply:
So when you step into the setFreq() function, the data is fine until some event happens? What sort of event will change the value?

Reply:
`S_config.freq` here is fine (say 2230.5). But in `setFreq()`, `S_freq_new` is 3.18e-18 something. Nothing really happens in between. But let's say you set other parameters, too, and then send another command. `S_config.freq` keeps its value but the rest `S_config.PA_Power` for example is back to its default value. So, it seems that the member `freq` is acting different than other members in `S_config`.

Reply:
All right. I have no idea what's happening lol! I was able to fix the problem (on the surface at least) by changing this to `status = HAL_S_setFreq(&S_config.freq);`. However, it didn't make any difference if the function is defined `HAL_S_setFreq( float x)` or `HAL_S_setFreq(float * x)`! But that's not the weird part! The thing I can't understand happens in `set_config`...

Reply:
@arrooney Just wanna make sure you've seen these comments. I could push a new commit but I'm in the middle of writing for UHF service and it's incomplete. So, let me know if that'd make things easier.